OverlayPlugin.Core/MemoryProcessors/Target/TargetMemoryManager.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections.Generic;
+
+namespace RainbowMage.OverlayPlugin.MemoryProcessors.Target
+{
+    public interface ITargetMemory
+    {
+        Combatant.Combatant GetTargetCombatant();
+
+        Combatant.Combatant GetFocusCombatant();
+
+        Combatant.Combatant GetHoverCombatant();
+    }
+
+    class TargetMemoryManager : ITargetMemory
+    {
+        private readonly TinyIoCContainer container;
+        private readonly FFXIVRepository repository;
+        private TargetMemory memory = null;
+
+        public TargetMemoryManager(TinyIoCContainer container)
+        {
+            this.container = container;
+            container.Register(new TargetMemory60(container));
+            container.Register(new TargetMemory61(container));
+            container.Register(new TargetMemory62(container));
+            repository = container.Resolve<FFXIVRepository>();
+        }
+
+        private void FindMemory()
+        {
+            List<TargetMemory> candidates = new List<TargetMemory>();
+            // For CN/KR, try the lang-specific candidate first, then fall back to intl

```

Comment:
I don't like the thought of checking unneeded candidates because each sigscan takes 1500ms approximately. Even with tying to process change, that's 20-30 seconds of CPU being pinned every time the process changes.

Reply:
That's fair, but I'd argue that in the worst case it's better to spend 20 seconds and have it work than to not have it work until the next update.  I think process change is very infrequent, so doing it once isn't so bad.  My concerns here are (majorly) it'd be nice to be forward-compatible to remove release pressure and just keep things working as versions move forward for the Chinese and Korean versions and (minorly) this pattern is repeated everywhere, so it'd be nice to not do that.

One compromise here is to add an interface to all the memory classes that say what version they are on, and we could make a sorter that walks an arbitrary list of candidates and sorts them in order of most likely.

And again, I think this is definitely a thought for the future and not for this patch.